src/packetizer/mod.rs

```diff
@@ -54,26 +35,20 @@ pub trait Depacketizer {
     fn is_partition_tail(&self, marker: bool, payload: &Bytes) -> bool;
 }
 
-//TODO: SystemTime vs Instant?
-// non-monotonic clock vs monotonically non-decreasing clock
-/// FnTimeGen provides current SystemTime
-pub type FnTimeGen =
-    Arc<dyn (Fn() -> Pin<Box<dyn Future<Output = SystemTime> + Send + 'static>>) + Send + Sync>;
-
 #[derive(Clone)]
-pub(crate) struct PacketizerImpl {
+pub struct RtpPacketizer<P, S> {
```

Comment:
Hmm, fair point. Especially the last bit around consumer control to still have dynamic dispatch.

Reply:
Actually that's slightly incorrect - I forgot about the weird quirk (well it's not that weird but unexpected) of the type system where `Box<dyn Trait>` doesn't actually implement `Trait`, so if you directly try to substitute in `Box<dyn Packetizer>` then it won't work but if you provide your own `DynPacketizer` type of sorts then it will work. For what its worth when we refactored our code to incorporate these changes, all we had to do was remove the calls to `Box::new` after making a packetizer and sequencer.

Reply:
Fair